tests/src/test/java/alluxio/master/file/ConcurrentFileSystemMasterTest.java

```diff
@@ -242,6 +309,27 @@ public void sameDirConcurrentRename() throws Exception {
   }
 
   /**
+   * Tests that many threads concurrently deleting the same directory will only succeed once.
+   */
+  @Test
+  public void sameDirConcurrentDelete() throws Exception {
```

Comment:
Can you also add a test to delete files under different directories with shared prefixes like /a/b/c, /a/b/d, /a/e?